pallets/asset-registry/src/tests.rs

```diff
@@ -29,84 +29,189 @@ const STATEMINE_ASSET_MULTI_LOCATION: MultiLocation = MultiLocation {
 	),
 };
 
-#[test]
-fn register_reserve_asset_works() {
-	new_test_ext().execute_with(|| {
-		assert_ok!(AssetRegistry::register_reserve_asset(
-			RuntimeOrigin::root(),
-			LOCAL_ASSET_ID,
-			STATEMINE_ASSET_MULTI_LOCATION,
-		));
-
-		assert_eq!(
-			AssetIdMultiLocation::<Test>::get(LOCAL_ASSET_ID),
-			Some(STATEMINE_ASSET_MULTI_LOCATION)
-		);
-		assert_eq!(
-			AssetMultiLocationId::<Test>::get(STATEMINE_ASSET_MULTI_LOCATION),
-			Some(LOCAL_ASSET_ID)
-		);
-	});
-}
-
-#[test]
-fn cannot_register_unexisting_asset() {
-	new_test_ext().execute_with(|| {
-		let unexisting_asset_id = 9999;
+mod register_reserve_assest {
+	use super::*;
 
-		assert_noop!(
-			AssetRegistry::register_reserve_asset(
+	#[test]
+	fn register_reserve_asset_works() {
+		new_test_ext().execute_with(|| {
+			assert_ok!(AssetRegistry::register_reserve_asset(
 				RuntimeOrigin::root(),
-				unexisting_asset_id,
+				LOCAL_ASSET_ID,
 				STATEMINE_ASSET_MULTI_LOCATION,
-			),
-			Error::<Test>::AssetDoesNotExist
-		);
-	});
-}
+			));
+
+			assert_eq!(
+				AssetIdMultiLocation::<Test>::get(LOCAL_ASSET_ID),
+				Some(STATEMINE_ASSET_MULTI_LOCATION)
+			);
+			assert_eq!(
+				AssetMultiLocationId::<Test>::get(STATEMINE_ASSET_MULTI_LOCATION),
+				Some(LOCAL_ASSET_ID)
+			);
+		});
+	}
+
+	#[test]
+	fn cannot_register_unexisting_asset() {
+		new_test_ext().execute_with(|| {
+			let unexisting_asset_id = 9999;
+
+			assert_noop!(
+				AssetRegistry::register_reserve_asset(
+					RuntimeOrigin::root(),
+					unexisting_asset_id,
+					STATEMINE_ASSET_MULTI_LOCATION,
+				),
+				Error::<Test>::AssetDoesNotExist
+			);
+		});
+	}
 
-#[test]
-fn cannot_double_register() {
-	new_test_ext().execute_with(|| {
-		assert_ok!(AssetRegistry::register_reserve_asset(
-			RuntimeOrigin::root(),
-			LOCAL_ASSET_ID,
-			STATEMINE_ASSET_MULTI_LOCATION,
-		));
-
-		assert_noop!(
-			AssetRegistry::register_reserve_asset(
+	#[test]
+	fn cannot_double_register() {
+		new_test_ext().execute_with(|| {
+			assert_ok!(AssetRegistry::register_reserve_asset(
 				RuntimeOrigin::root(),
 				LOCAL_ASSET_ID,
 				STATEMINE_ASSET_MULTI_LOCATION,
+			));
+
+			assert_noop!(
+				AssetRegistry::register_reserve_asset(
+					RuntimeOrigin::root(),
+					LOCAL_ASSET_ID,
+					STATEMINE_ASSET_MULTI_LOCATION,
+				),
+				Error::<Test>::AssetAlreadyRegistered
+			);
+		});
+	}
+
+	#[test]
+	fn valid_locations_succced() {
```

Comment:
can we add one more tests to validate relay chain native token?

`MultiLocation { parents: 1, interior: Here `

Reply:
Using `Here` will not work since the check is based on how this is suggested by the XCM-Format.

> chain/PalletInstance(<id>) for a Frame chain with a single-asset pallet instance (such as an instance of the Balances pallet).

In order for that to work it would need to be `MultiLocation { parents: 1, interior: PalletInstance(4)}` where the id 4 references the rococo pallet balances.

So I'm not sure whether we should tweak the validation or maybe I`m missing something

Reply:
But in PolkadotJS we are allowed to choose `Here` for the Concrete identifiers.

![Screenshot 2023-06-10 at 08 24 25](https://github.com/paritytech/trappist/assets/12089572/28401669-c241-452f-abbe-b7b9ad886658)

My guess is that it should work

Reply:
It's fixed with the commit I added yesterday.
```rust
        let (split_multilocation, last_junction) = location.clone().split_last_interior();

	let check = matches!(
		last_junction,
		Some(AccountId32 { .. }) |
			Some(AccountKey20 { .. }) |
			Some(PalletInstance(_)) |
			None
	);
```
If the Junctions is `Here` the function `split_interior()` will return `None`.